app/javascript/App/components/NewsfeedPage/CoinList.js

```diff
@@ -1,29 +1,58 @@
-import React, { Fragment } from 'react'
+import React, { Component, Fragment } from 'react'
 import CoinListHeader from './CoinListHeader'
 import CoinListItem from './CoinListItem'
 import LoadingIndicator from '../LoadingIndicator'
+import coinSearchProvider from '../../containers/coinSearch'
 
-export default (props) => {
-  const { isLoading, setActiveEntity, currentUI, isWatching } = props
-  const onClick = (coin) =>
-    setActiveEntity({
+class CoinList extends Component {
+  onClickCoin = (coin) =>
```

Comment:
IMHO this function could have a more meaningful name. instead of calling it what happens, 'on click coin', it may be more readable calling it by what it is going to do, 'set active'. then the user can derive what it's doing and see see this in the callstack. 

so instead of looking like: 
```
onClick={onClickCoin}
onClickCoin = (coin) => setActiveEntity(coin)
```

it would look like this: 
```
onClick={setActive}
setActive = (coin) => setActiveEntity(coin)
```

then later you have another handler, that's preparing the ui by clearing the search and invoking toggleUI. 

instead of looking like: 

```
onClickNewCoin = (coin) => {
  this.onClickCoin(coin)
```

this could look like: 

```
newCoinHandler = (coin) => {
  this.setActive(coin)
```

Reply:
Makes sense :)